sockeye/beam_search.py

```diff
@@ -622,21 +624,24 @@ def forward(self,
         # target vocab for this sentence.
         if restrict_lexicon:
             source_words = source[:, :, 0]
-            vocab_slice_ids, _ = _get_vocab_slice_ids(restrict_lexicon, source_words, self.eos_id, beam_size=1, target_prefix=target_prefix)
+            vocab_slice_ids, _ = _get_vocab_slice_ids(restrict_lexicon, source_words, self.eos_id,
+                                                      beam_size=1, target_prefix=target_prefix)
 
         # (0) encode source sentence, returns a list
         model_states, _ = self._inference.encode_and_initialize(source, source_length)
         # TODO: check for disabled predicted output length
 
         # Prefix masks, where scores are infinity for all other vocabulary items except target_prefix ids
-        prefix_masks, prefix_masks_length = (utils.gen_prefix_masking(target_prefix, self.output_vocab_size, self.dtype), target_prefix.size(1)) \
-                                            if target_prefix is not None else (None, None)  # type: ignore
-        if prefix_masks is not None and vocab_slice_ids is not None:
-            prefix_masks = pt.index_select(prefix_masks, -1, vocab_slice_ids)
+        prefix_masks, prefix_masks_length = None, 0
+        if prefix_masks is not None:
```

Comment:
Hi, something off here I think (prefix_masks is assigned to None in 635).

Reply:
Should be: if target_prefix is not None

Reply:
good cathch, thanks!